packages/material-ui/src/TextField/TextField.js

```diff
@@ -1,13 +1,16 @@
 // @inheritedComponent FormControl
 
 import React from 'react';
+import ReactDOM from 'react-dom';
+import ResizeObserver from 'resize-observer-polyfill';
```

Comment:
Why do we have to change the text field. Do it mean people can't compose the form components?

Reply:
What can we do to remove this dependency?

Reply:
I originally tried doing this at the form level, and that would be the ideal case, but the amount of styling edge cases and the overhead needed to track resizing on the svg made that a no-go from everything I tried.

We need `ref`s to the `input` and the `label` to size the `NotchedOutline`, which is positioned `relative`. Many of these difficult cases come about because the svg is not a proper border or container for the input element.

On `resize-observer-polyfill`, I'm not sure how to do so while maintaining render performance and quality. The main approaches I've seen are listening to `window` `resize` events, but this doesn't account for other size changes that are unrelated to the `window` size, and fires more frequently than is desirable. We could throttle this, but still I don't think it cleanly resolves the resize of the `fullWidth` or `multiline` inputs. The final approach I've considered is a long-lived `setInterval` of around `10Hz` (the frequency I've seen in a couple other places), but this would have a performance impact, and is probably a last resort.

@oliviertassinari what do you feel is best?

Reply:
Interesting use case of this dependency: https://github.com/palantir/blueprint/pull/2718

Reply:
That's a good reference to have on my end as well, if we decide to take this approach, thanks for linking that. I haven't yet accounted for `componentDidUpdate(...)`

Reply:
By using a white CSS border, we can keep the TextField a dump wrapper.